app.test.js

```diff
@@ -108,12 +107,55 @@ describe("GET: /api/articles", () => {
       expect(body).toBeSortedBy('created_at', {descending: true})
     })
   })
-  test("Return bad request if url is incorrect", () => {
+  test("Return not found if url is incorrect", () => {
     return request(app)
     .get('/api/articulz')
     .expect(404)
     .then((body) => {
     expect(body.notFound).toBe(true)
   })
 })
-})
\ No newline at end of file
+})
+
+describe("GET: /api/articles/:article_id/comments", () => {
+  test("Return an array of comments from specified article", () => {
+  return request(app)
+  .get('/api/articles/9/comments')
+  .expect(200)
+  .then(({body}) => {
+    expect((body.article.length)).toBe(2)
+    expect((body.article[0].article_id)).toBe(9)
+    expect((body.article[1].article_id)).toBe(9)
+    expect(body.article).toBeSortedBy('created_at', {descending: true})
+  })
+})
+  test("Return 404 error if article ID doesnt exist", () => {
+    return request(app)
+    .get('/api/articles/420420/comments')
+    .expect(404)
+    .then((body) => {
+      expect(body.notFound).toBe(true)
+    })
+  })
+  
+  describe("POST: /api/articles/:article_id/comments", () => {
+    test("Return new comment to specified article", () => {
+      return request(app)
+      .post('/api/articles/9/comments')
+      .send({body: "Hello", username: "icellusedkars"})
+      .expect(201)
+      .then(({body}) => {
+        expect((expect((body.body)).toBe("Hello")))
+        expect((body.author)).toBe("icellusedkars")
+      })
+    })
+    test("Return not found if id is incorrect", () => {
+      return request(app)
+      .get('/api/articules/3945/comments')
+      .expect(404)
+      .then((body) => {
+      expect(body.notFound).toBe(true)
+    })
+    })
+  })
```

Comment:
Great tests so far, however there are some cases which have not been tested for yet. What should happen if a request was made that was missing some of the required fields in the request body? i.e. no username or body properties❓ 

You want to add a test for a 201 - that ignores unnecessary properties.
Add another 404 for when a username doesn't exist and a 400 for invalid ID e.g. string of "not-an-id"